phoenix-core/src/main/java/org/apache/phoenix/parse/IndexExpressionParseNodeRewriter.java

```diff
@@ -51,6 +51,12 @@ public Void visit(ColumnParseNode node) throws SQLException {
             return null;
         }
         
+        @Override
+        public boolean visitEnter(FunctionParseNode node) throws SQLException {
```

Comment:
Is the intent of this visitor to determine if the indexed column is solely a column name that's not case sensitive? Seems like you'd want to switch the starting value in the constructor of isParseNodeCaseSensitive to true and then only set it false in the visit(ColumnParseNode) if isTopLevel() is true (i.e. it's the only expression in the tree) and the node isn't already case sensitive.